### PR TITLE
Update documentation architecture

### PR DESCRIPTION
**Pull request by wbeuil**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | none
| ❓ Documentation  | yes

## Description

Remove duplicate doc files, update root README and re-organize docs folder.

After this PR, we should have in mind that our documentation is in the other repo cloudskiff/driftctl-docs.

Only developer related docs will remain in this repo (e.g. contributing).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,169 +40,23 @@ driftctl tracks how well your IaC codebase covers your cloud configuration. drif
...
-## Documentation & support
+---
 
-- [Get started](https://driftctl.com/product/quick-tutorial/)
-- [User guide](doc/README.md)
-- [Discord](https://discord.gg/NMCBxtD7Nd)
+**[Get Started](https://driftctl.com/product/quick-tutorial/)**
 
-## Getting started
+**[Documentation](https://docs.driftctl.com)**
```
Comment (eliecharra) [inline README.md:51]: It could be nice to have a direct link to installation page there

**Review by eliecharra — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (wbeuil) [reply README.md:51]: Do you mean I need to add after `Documentation` a link to `Installation` ? Or simply change the URL of the `Documentation` to redirect to the `Installation` page ?

Comment (eliecharra) [reply README.md:51]: I meant a new link after or maybe before documentation

Comment (wbeuil) [reply README.md:51]: Done fca36e1

**Comment by eliecharra** [comment]
LGTM overall, a conflict need to be fixed, maybe we should wait all new resources PRs to be merged before rebasing this one

**Review by moadibfr — DISMISSED** [review]

**Comment by wbeuil** [comment]
Documentation updated on driftctl-docs repo

**Review by eliecharra — APPROVED** [review]

**Pull request merged**